### DISCUSSION TITLE
Support for Trino interface

### DISCUSSION THREAD

**Discussion by iavrekh** [xoolive/traffic#433] [Q&A]
Please advise which version of Traffic library supports Trino interface. I've been using 2.6.1 for quite a while and, apparently, can no longer pull historical data from OpenSky (I was informed by OpenSky about the change recently and set a new account per their request).

**Comment by xoolive** [comment]
Try the latest

**Reply by iavrekh** [reply]
Thank you for the quick response!
I'll try to do that.

**Reply by iavrekh** [reply]
Following your suggestion, I've installed the latest Traffic development code and was able to download historical data (daily batch of flights in the bounding rectangle around KSFO, roughly representing the terminal airspace) w/out changing my code, which is using "opensky.history" method. To validate the result, I've downloaded the data for the same day that I've downloaded earlier using "Impala interface" and put it through my processing "pipeline", which separates arriving and departing flights and further characterizes them e.g. by landing runway, final approach procedure and overall approach stability. The results came out remarkably similar in terms of the number of arriving and departing flights and stability characteristics. However, there are some noticeable differences - "Trini flights", as a rule, have longer trajectories and more time points even for trajectories of the same duration (ergo, significantly larger download files). I've also noticed some differences in geoaltitude and vertical rate values at the same time point for the same flight retrieved by Impala and Trini interfaces. Does it mean that we are dealing with two different databases? I don't think it's necessarily bad, I just need to know whether I'll, eventually, need to reload/reprocess the data, which I've already downloaded through the legacy interface.

**Comment by iavrekh** [comment]
Apologies for my previous reply. After careful examination I was able to establish that the data retrieved by the newer version of Traffic library using Trino interface is completely correct. The differences I mentioned earlier were due to the application of the different filtering algorithms during post-processing.
